src/psbt.c

```diff
@@ -2201,3 +2203,733 @@ int wally_psbt_to_base64(
+
+            if (!input->partial_sigs || input->partial_sigs->num_items < n_sigs) {
+                continue;
+            }
+
+            if (!script_is_op_n(out_script[out_script_len - 2], false, &n_pks)) {
+                // How did this happen?
+                return WALLY_ERROR;
+            }
+
+            sigs_len = EC_SIGNATURE_LEN * n_sigs;
+            if (!(sigs = wally_malloc(sigs_len)) || !(sighashes = wally_malloc(n_sigs * sizeof(uint32_t)))) {
+                return WALLY_ENOMEM;
+            }
+
+            // Go through the multisig script and figure out the order of pubkeys
+            p++; // Skip the n_sig item
+            for (j = 0; j < n_pks && p < end; ++j) {
+                size_t push_size, push_opcode_size, sig_len;
+                unsigned char *pubkey, *sig, compact_sig[EC_SIGNATURE_LEN];
+                bool found = false;
+
+                if ((ret = script_get_push_size_from_bytes(p, end - p, &push_size)) != WALLY_OK) {
+                    wally_free(sigs);
+                    wally_free(sighashes);
+                    return ret;
+                }
+                if ((ret = script_get_push_opcode_size_from_bytes(p, end - p, &push_opcode_size)) != WALLY_OK) {
+                    wally_free(sigs);
+                    wally_free(sighashes);
+                    return ret;
+                }
+                p += push_opcode_size;
+
+                pubkey = p;
+                p += push_size;
+
+                for (k = 0; k < input->partial_sigs->num_items; ++k) {
+                    if (memcmp(input->partial_sigs->items[k].pubkey, pubkey, push_size) == 0) {
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found) {
+                    continue;
+                }
+
+                // Get the signature and sighash separately
+                sig = input->partial_sigs->items[k].sig;
+                sig_len = input->partial_sigs->items[k].sig_len; // Has sighash byte at end
+                if ((ret = wally_ec_sig_from_der(sig, sig_len - 1, compact_sig, EC_SIGNATURE_LEN)) != WALLY_OK) {
+                    wally_free(sigs);
+                    wally_free(sighashes);
+                    return ret;
+                }
+                memcpy(sigs + sig_i * EC_SIGNATURE_LEN, compact_sig, EC_SIGNATURE_LEN);
+                sighashes[sig_i] = (uint32_t)sig[sig_len - 1];
+                sig_i++;
+            }
+
+            if (witness) {
+                if ((ret = wally_witness_multisig_from_bytes(out_script, out_script_len, sigs, sigs_len, sighashes, n_sigs, 0, &input->final_witness)) != WALLY_OK) {
+                    wally_free(sigs);
+                    wally_free(sighashes);
+                    return ret;
+                }
+            } else {
+                script_sig_len = n_sigs * (EC_SIGNATURE_DER_MAX_LEN + 2) + out_script_len;
+                script_sig = wally_malloc(script_sig_len);
```

Comment:
nit: Maybe explicitly check for alloc failure otherwise I think we'll return WALLY_EINVAL rather than WALLY_ENOMEM

Reply:
Done